### PR TITLE
[PIN-93] [PIN-103] Map legend

### PR DESCRIPTION
**Pull request by mxj220**
- Added colored pins for different pin sources
- Changed cloud functions to store pin source
- Added map legend
- Added filtering by pin source

### CODE REVIEW

File: app/src/main/java/edu/wisc/ece/pinpoint/MainActivity.java [hunk 1]
```diff
@@ -56,10 +63,15 @@ public void onCreate(Bundle savedInstanceState) {
...
-        firebase.fetchActivity(uid);
+        // List of fetching tasks that must be completed before launching app content
+        List<Task<Object>> fetchTasks = new ArrayList<>();
+        // Tasks must be continued with Object tasks to be added to fetch list
+        fetchTasks.add(firebase.fetchUser(uid).continueWith(t -> null));
+        fetchTasks.add(firebase.fetchFollowing(uid).continueWith(t -> null));
+        fetchTasks.add(firebase.fetchFollowers(uid).continueWith(t -> null));
+        fetchTasks.add(firebase.fetchActivity(uid).continueWith(t -> null));
+        // Wait until all tasks complete before showing view
+        Tasks.whenAllComplete(fetchTasks).addOnCompleteListener(fetchingComplete -> showView(R.id.content_view));
```
Comment (skovati) [inline app/src/main/java/edu/wisc/ece/pinpoint/MainActivity.java:74]: This is so clean

File: app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java [hunk 2]
```diff
@@ -111,6 +125,69 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+            }
+            else {
+                filterContainer.animate().translationX(-160 * scale + 0.5f);
+                isFilterVisible = true;
+                filterTab.setImageResource(R.drawable.ic_filter);
+            }
+        });
+
+        generalBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            for(Marker marker : generalMarkers){
```
Comment (skovati) [inline app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java]: Maybe the following could be cleaner?
```java
droppedMarkers.forEach(m -> m.setVisible(isChecked));
```

File: app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java [hunk 3]
```diff
@@ -179,8 +275,22 @@ private void handleUndiscoveredPinClick(Marker marker) {
...
+                PinSource pinSource;
+                switch (marker.getSnippet()){
+                    case "DEV":
+                        pinSource = PinSource.DEV;
+                        break;
+                    case "NFC":
+                        pinSource = PinSource.NFC;
+                        break;
+                    default:
+                        // Do not store friend enum in cloud,
```
Comment (skovati) [inline app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java:287]: Do we want to be storing NFC status in the cloud either?

**Review by skovati — APPROVED** [review]
Looks CLEAN :heart_eyes:

File: app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java (hunk 3 above)
Comment (mxj220) [reply app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java:274]: I'll find out what works best for nfc in my next ticket

**Pull request merged**